yapapi/services/service.py

```diff
@@ -365,20 +361,19 @@ async def reset(self) -> None:
         Handlers of a restarted service are called more then once - all of the cleanup necessary between calls
         should be implemented here. E.g. if we initialize a counter in :func:`Service.__init__` and increment it
         in :func:`Service.start()`, we might want to reset it here to the initial value.
-
-        Target implementation (0.10.0 and up) will raise NotImplementedError. Current implementation only warns about
-        this future change.
         """
+        pass
 
-        msg = (
-            "Default implementation of Service.reset will raise NotImplementedError starting from 0.10.0. "
-            f"You should implement this method on {type(self)}"
+    @property
+    def restart_condition(self) -> bool:
+        """Dictated condition based on which this instance will be restarted."""
```

Comment:
Again, we should mention the relationship between `restart_condition` and `reset`